<launch>

  <arg name="dof"                               default="4"/>
  <arg name="robot_model"                       default="px100"/>
  <arg name="world_name"                        default="$(find px100_simulation)/worlds/px100-0.world"/>
  <arg name="table_height"                      default="0.715"/>
  <arg name="x_offset"                          default="-0.15"/>
  

  <!-- Set up simulation world -->
  <include file="$(find px100_simulation)/launch/gazebo.launch">
    <arg name="robot_model" value="$(arg robot_model)" />
    <arg name="world_name" value="$(arg world_name)" />
    <arg name="table_height" value="$(arg table_height)" />
    <arg name="x_offset" value="$(arg x_offset)" />
  </include>
  
  
  <!-- Publish statics and visuals of robot arm via .urdf.xacro -->
  <include file="$(find px100_simulation)/launch/description.launch" />
  
  
  <!-- Publish dynamics for the robot, tune PID controls here -->
  <rosparam file="$(find px100_simulation)/config/controllers.yaml" command="load" ns="$(arg robot_model)"/>
  <node pkg="controller_manager" name="controller_spawner" type="controller_manager" respawn="false" output="screen"
    ns="$(arg robot_model)" args="spawn arm_controller gripper_controller joint_state_controller"/>


  <!-- Launch moveit planning integration -->
  <!--<include file="$(find interbotix_xsarm_moveit)/launch/move_group.launch" ns="$(arg robot_model)">-->
  <!--  <arg name="robot_model"                       value="$(arg robot_model)"/>-->
  <!--  <arg name="dof"                               value="$(arg dof)"/>-->
  <!--  <arg name="fake_execution"                    value="false"/>-->
  <!--</include>-->
  

  <!-- Start streaming camera data to KVS -->
  <!--<include file="$(find px100_simulation)/launch/kvstreamer.launch" ns="$(arg robot_model)" />-->


  <!-- Open Rviz visualiztion for the system -->
  <node name="rviz" pkg="rviz" type="rviz" respawn="false" output="screen"
    args="-f world -d $(find px100_simulation)/rviz/px100.rviz" ns="$(arg robot_model)">
  </node>
  
  
  <!-- Start image view node for taking camera shots -->
  <node name="image_saver" pkg="image_view" type="image_saver" output="screen" ns="$(arg robot_model)">
    <remap from="image" to="camera/image_raw" />
    <param name="save_all_image" value="false" />
    <param name="filename_format" value="$(find px100_simulation)/nodes/image_%04d.png" />
  </node>
  
  
  <!-- Set robot to initial position -->
  <node pkg="px100_simulation" name="init" type="init.py" output="screen" />
  
</launch>
